carbon-design-system/carbon
# feat(component): replaced `SideNavItem.js` -> `SideNavItem.tsx` component

Closes #13602 

To add a new file `SideNavItem.tsx` with it's corresponding types.

#### Changelog

**New**

- `SideNavItem.tsx`

**Changed**

- Updated TypeScript types for the SideNavItem component to resolve type error.

**Removed**

- None

#### Testing / Reviewing

To verify the changes:

1. Ensure that the SideNavItem component renders without any TypeScript errors.
2. Test the component with different props and verify that it behaves as expected.
3. Check if the component integrates seamlessly with the rest of the application's components and features.
4. Run existing tests to ensure that the changes do not break any existing functionality.
